Cod3rsGrowth.Web/wwwroot/Webapp/app/carros/AdicionarCarro.controller.js

```diff
@@ -0,0 +1,283 @@
+    const parametroArgumento = "arguments";
+    const idDoInputMarca = "SelecionarMarca";
+    const rotaEditarCarro = "appEditarCarro";
+    const rotaCriarCarro = "appAdicionarCarro";
+    const rotaListagemCarros = "appListagemCarro";
+    const url= "http://localhost:5071/api/Carros/";
+    const idTituloViewCriarCarro = "TituloEditarCarro";
+    const textoParaAdicionarTituloEdicao = "Editar Carro";
+    const textoParaAdicionarTituloCriacao = "Adicionar Carro";
+    const urlCores = "http://localhost:5071/api/Carros/Cores";
+    const idDoMessageStripErroCriarCarro = "erroAoCriarCarro";
+    const urlMarcas = "http://localhost:5071/api/Carros/Marcas";
+    const idDoMessageStripErroEditarCarro = "erroAoEditarCarro";
+    const idDoMessageStripSucessoCriacao = "sucessoAoCriarCarro";
+    const idDoMessageStripSucessoEditar = "sucessoAoEditarCarro";
+
+    return BaseController.extend("ui5.carro.app.carros.AdicionarCarro", {
+        formatter: Formatter,
+
+        onInit() {
+            this.aoCoincidirRota();
+        },
+
+        aoCoincidirRota() {
+            this.processarEvento(() => {
+                var rota = sap.ui.core.UIComponent.getRouterFor(this);
+                rota.getRoute(rotaCriarCarro).attachMatched(this._carregarEventosCriar, this);
+                rota.getRoute(rotaEditarCarro).attachMatched(this._carregarEventodEditar, this);
+            });
+        },
+
+        _carregarEventodEditar(oEvent) {
+            this._mudarTituloDaViewEdicao();
+            this._removerMessageStrip();
+            this._limparInputs();
+            this._obterCarroPorId(oEvent);
+            this._carregarDescricaoCores();
+            this._carregarDescricaoMarcas();
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+        },
+
+        _carregarEventosCriar(oEvent) {
+            this._mudarTituloDaViewCriar();
+            this._limparInputs();
+            this._removerMessageStrip();
+            this._carregarDescricaoCores();
+            this._carregarDescricaoMarcas();
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+        },
+
+        _limparInputs() {
+            const inputModelo = this.oView.byId(idDoInputModelo);
+            const modelo = inputModelo.setValue(null);
+            inputModelo.setValueState(sap.ui.core.ValueState.None);
+            inputModelo.setValueStateText('');
+            const inputValor = this.oView.byId(idDoInputValor);
+            const valor = inputValor.setValue(null);
+            inputValor.setValueState(sap.ui.core.ValueState.None);
+            inputValor.setValueStateText('');
+            const flex = this.oView.byId(idDoInputFlex).setState(false)
+        },
+
+        _removerMessageStrip(){
+            this.getView().byId(idDoMessageStripErroCriarCarro).setVisible(false);
```

Comment:
_removerMessageStrip() {
    const messageStripIds = [
        idDoMessageStripErroCriarCarro,
        idDoMessageStripErroEditarCarro,
        idDoMessageStripSucessoCriacao,
        idDoMessageStripSucessoEditar
    ];

    messageStripIds.forEach(id => {
        this.getView().byId(id).setVisible(false);
    });
}